### PR TITLE
Dynamic Delay: CFs wont override each other (#358)

### PR DESCRIPTION
**Pull request by Yuval-Ariel**
Currently, in a setting of multiple cfs, the delayed_write_rate_ value is overridden by the latest cf that entered a delay state. so if a cf registered a delay state that is less severe than the previous cf, it will override the previous value of delayed_write_rate_ with a higher rate than the system should use.

fix it by storing the delay requirements of all the CFs in the write controller and only set a new delayed_write_rate_ when its the min value from all entries stored.

### CODE REVIEW

File: db/write_controller.h [hunk 1]
```diff
@@ -50,6 +50,32 @@ class WriteController {
   // threads will be increased
   std::unique_ptr<WriteControllerToken> GetCompactionPressureToken();
 
+  struct RateAndWasMin {
```
Comment (ayulas) [inline db/write_controller.h:53]: why not using set that the cf pointer is the element (it consists the write rate and id ) and the sorted set function is the write rate so the min is the first element. that way when you when you remove / change/add the you just need call set with the first set element write rate...
you dont need the is min etc ... you dont need all above functions...

File: db/write_controller.cc [hunk 2]
```diff
@@ -33,6 +33,50 @@ std::unique_ptr<WriteControllerToken> WriteController::GetDelayToken(
   return std::unique_ptr<WriteControllerToken>(new DelayWriteToken(this));
 }
 
+void WriteController::InsertOrAssignToCfIdAndRateMap(uint32_t cf_id,
```
Comment (ayulas) [inline db/write_controller.cc:36]: see my buttom comments. if you use std set with cf object pointer you dont need this

**Review by ayulas — DISMISSED** [review]
we discussed , and for now its good enough . for future we might think of map  with custom compare

**Comment by Yuval-Ariel** [comment]
i've considered replacing the logic for finding the min rate inside GetMinRate() with std::min_element but i'm setting is_min = false on all elements through iteration anyways..

**Comment by Yuval-Ariel** [comment]
reabased on 2.2.1 since there were major db bench changes then and the recent baseline in on 2.2.1. @ayulas , plx reapprove

**Pull request closed without merging**